src/Data/ProgrammesDb/EntityRepository/CollapsedBroadcastRepository.php

```diff
@@ -167,6 +167,24 @@ public function findByCategoryAncestryAndStartAtDateRange(
         return $this->resolveProgrammeParents($result);
     }
 
+    public function findByProgramme(
+        array $ancestry,
+        bool $isWebcastOnly,
+        ?int $limit,
+        int $offset
+    ): array {
+        $qb = $this->createCollapsedBroadcastsOfProgrammeQueryBuilder($ancestry, $isWebcastOnly)
+            ->addOrderBy('collapsedBroadcast.endAt', 'ASC')
+            ->setFirstResult($offset)
+            ->setMaxResults($limit);
+
+        $result = $qb->getQuery()->getResult(Query::HYDRATE_ARRAY);
+
+        $result = $this->explodeFields($result);
+
+        return $this->resolveProgrammeParents($result);
```

Comment:
Same thing: do you actually need the whole ancestry resolved?

Reply:
The ancestry should already be cached in memory at this point, as it will inevitably consist of the episode this is a broadcast of and its parents, which will already have been queried.